app/code/Magento/Backend/view/adminhtml/web/js/store-switcher.js

```diff
@@ -0,0 +1,88 @@
+define([
+    'jquery',
+    'mage/translate',
+    'Magento_Ui/js/modal/confirm'
+], function($, $t, confirm) {
+    'use strict';
+
+    return function(config) {
+        var scopeSwitcherHandler;
+
+        (function($) {
+            var storesList = $('[data-role=stores-list]');
+            storesList.on('click', '[data-value]', function(event) {
+                var val = $(event.target).data('value');
+                var role = $(event.target).data('role');
+                var switcher = $('[data-role='+role+']');
+
+                event.preventDefault();
+
+                if (!switcher.val() || val != switcher.val()) {
+                    /* Set the value & trigger event */
+                    switcher.val(val).trigger('change');
+                }
+            });
+        })($);
+
+        function switchScope(obj) {
+            var switcher = $(obj);
+            var scopeId = switcher.val();
+            var scopeParams = '';
+            if (scopeId) {
+                scopeParams = switcher.data('param') + '/' + scopeId + '/';
+            }
+
+            if (obj.switchParams) {
+                scopeParams += obj.switchParams;
+            }
+
+            if (typeof scopeSwitcherHandler != 'undefined') {
+                var switcherParams = {
+                    scopeId: scopeId,
+                    scopeParams: scopeParams,
+                    useConfirm: config.useConfirm
+                };
+                scopeSwitcherHandler(switcherParams);
+            } else {
+                if (config.useConfirm) {
+                    confirm({
+                        content: $t('Please confirm scope switching. All data that hasn\'t been saved will be lost.'),
+                        actions: {
+                            confirm: function () {
+                                reload();
+                            },
+                            cancel: function () {
+                                obj.value = config.storeId ? config.storeId : '';
+                            }
+                        }
+                    });
```

Comment:
maybe that's a bit stupid question, but can we wrap it to require and load `'mage/translate'` and `'Magento_Ui/js/modal/confirm'` components when this really needed?

Reply:
updated